src/containers/tutor-home-page/general-info-step/SelectGroup.tsx

```diff
@@ -1,30 +1,89 @@
-import React from 'react'
-import { FormControl, InputLabel, Select, MenuItem } from '@mui/material'
+import React, { useEffect, useState } from 'react'
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  SelectChangeEvent
+} from '@mui/material'
 import { useTranslation } from 'react-i18next'
 import { useSelectGroupStyles } from './SelectGroup.styles'
+import { LocationService } from '~/services/location-service'
 
 const SelectGroup: React.FC = () => {
   const { t } = useTranslation()
   const classes = useSelectGroupStyles()
+  const [countries, setCountries] = useState<string[]>([])
+  const [cities, setCities] = useState<string[]>([])
+  const [selectedCountry, setSelectedCountry] = useState('')
+  const [selectedCity, setSelectedCity] = useState('')
+
+  useEffect(() => {
+    const fetchCountries = async () => {
+      try {
+        const response = await LocationService.getCountries()
+        setCountries(response.data as string[])
+      } catch (error) {
+        console.error('Error fetching countries:', error)
+      }
+    }
+
+    void fetchCountries()
+  }, [])
+
+  useEffect(() => {
+    if (selectedCountry) {
```

Comment:
What happen if the `selectedCountry` becomes empty?

Reply:
Added the automatic clearance of the input and disable the city if the country is not selected